spec/configuration_spec.lua

```diff
@@ -125,6 +125,53 @@ describe('Configuration object', function()
       assert.same(services, filter_services(services, { '42' }))
       assert.same({}, filter_services(services, { '21' }))
     end)
+
+    describe("with service filter", function()
+      local mockservices = {
+        {id="42", hosts={"test.foo.com", "test.bar.com"}},
+        {id="12", hosts={"staging.foo.com"}},
+        {id="21", hosts={"prod.foo.com"}},
+      }
+
+      it("validates default case", function()
+        assert.same(filter_services(mockservices, nil), mockservices)
+      end)
+
+      it("reads from environment variable", function()
+        env.set('APICAST_SERVICES_FILTER_BY_URL', '.*.foo.com')
+        assert.same(filter_services(mockservices, nil), mockservices)
+
+        env.set('APICAST_SERVICES_FILTER_BY_URL', '^test.*')
+        assert.same(filter_services(mockservices, nil), {mockservices[1]})
+
+        env.set('APICAST_SERVICES_FILTER_BY_URL', '^(test|prod).*')
+        assert.same(filter_services(mockservices, nil), {mockservices[1], mockservices[3]})
+      end)
+
+      it("matches the second host", function()
+        env.set('APICAST_SERVICES_FILTER_BY_URL', '^test.bar.com')
+        assert.same(filter_services(mockservices, nil), {mockservices[1]})
+      end)
+
+      it("validates invalid regexp", function()
+        env.set('APICAST_SERVICES_FILTER_BY_URL', '^]')
+        assert.same(filter_services(mockservices, nil), {})
+      end)
+
+      it("combination with service list", function()
+        env.set('APICAST_SERVICES_FILTER_BY_URL', '^test.*')
+        assert.same(filter_services(mockservices, {"21"}), {
```

Comment:
Would this test be clearer if the 2 variables were set at the beginning and we had a single assert?